libpod/image/image.go

```diff
@@ -902,6 +930,9 @@ func (i *Image) MatchRepoTag(input string) (string, error) {
 		}
 	}
 	if maxCount == 0 {
+		if hasChildren {
+			return "", errors.Errorf("unable to delete %q (cannot be forced) - image has dependent child images", i.ID())
```

Comment:
This function otherwise has nothing to do with deletion, I don't think this makes sense here

Reply:
Moved it out of here. Fixed